src/cli/publish_command.zig

```diff
@@ -979,10 +998,8 @@ pub const PublishCommand = struct {
             .value = Expr.init(
                 E.String,
                 .{
-                    .data = try std.fmt.allocPrint(allocator, "http://{s}/{s}/-/{}", .{
-                        // always use replace https with http
-                        // https://github.com/npm/cli/blob/9281ebf8e428d40450ad75ba61bc6f040b3bf896/workspaces/libnpmpublish/lib/publish.js#L120
-                        strings.withoutTrailingSlash(strings.withoutPrefixComptime(registry.url.href, "https://")),
+                    .data = try std.fmt.allocPrint(allocator, "{s}/{s}/-/{}", .{
```

Comment:
@dylan-conway this should use the URL parser.. The protocol is unlikely to cause issues. But, the pathname could very much cause issues.